### PR TITLE
Convert @inertiajs/inertia to TypeScript 🚀

### PR DESCRIPTION
**Pull request by snaptopixel**
global changes:
- Add `lint` script(s) to each repo and base package.json
- Get eslint actually working and passing
- Autofix a few things

@inertiajs/inertia changes:
- Convert to TS and add typing
- Update build scripts and config for TS
- TWEAK: Don't use default exports (https://basarat.gitbook.io/typescript/main-1/defaultisbad)
- FEATURE: Create, use and export a custom Axios instance, so consumers can customize it as needed (fix/workaround for [this](https://github.com/inertiajs/inertia/issues/195))
- BUGFIX: `isBackForwardVisit` was referencing `type` when it should have been `entryType`

### CODE REVIEW

**Comment by snaptopixel** [comment]
@reinink this should be comprehensive, happy to discuss and answer any questions you may have. I'm also going to create a separate PR adding the generated types to the current lib, until you decide how to proceed with TypeScript.

**Comment by reinink** [comment]
Very cool @snaptopixel! I hope to have some time to dig into this early in the new year. 🙌

**Comment by snaptopixel** [comment]
@reinink ☝️ FYI I'll try and keep this up to date with master as you guys merge things.

File: packages/inertia/src/events.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+function fireEvent(name, options) {
```
Comment (snaptopixel) [inline packages/inertia/src/events.js]: Ignore this, I accidentally added this js file (and removed on the following commit)

**Comment by reinink** [comment]
Closing this, as Inertia core has now been converted to Typescript. Thanks for your help on this @snaptopixel!! 👏

**Comment by snaptopixel** [comment]
Awesome news @reinink thanks for the mention 😊

**Pull request closed without merging**